src/messaging/ExchangeMgr.cpp

```diff
@@ -236,11 +236,20 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
                         packetHeader.GetDestinationGroupId().Value());
     }
 
+    //
+    // If the session isn't active, we shouldn't proceed any further.
+    //
+    if (!session->IsActiveSession())
+    {
+        ChipLogError(ExchangeManager, "OnMessageReceived failed due to receiving a message on an in-active session");
+        return;
```

Comment:
Should it be the responsibility of the exchange manager to check this?  If we are already here, doesn't this mean that the session manager believes the session is viable and active?

Reply:
Would you rather we shift the checks higher into `SessionManager::OnMessageReceived`? That's likely the better place.

Reply:
Yes, I don't think the session manager should be forwarding anything if the session is not valid.

Related question: do we yet have the logic that will promote defunct sessions back to active when more messages are received?  If so, is this only for sessions that are marked for eviction?

Reply:
I wonder whether we should reply a StandaloneAck here. It is probably a gray area which is not covered by spec. No matter whether we do, it should align with the spec.

It is possible to replay a StandaloneAck if we make a change that allow EpheremalExchange being created on an inactive session.

Reply:
I think if we are rejecting the session, we must definitely not ack.